pkg/runtest/run.go

```diff
@@ -178,7 +183,8 @@ func (ctx *Context) generatePrograms() error {
 	}
 	for _, file := range files {
 		if err := ctx.generateFile(sandboxes, cover, file); err != nil {
-			return err
+			ctx.log("generateFile(%v) returned an error: %v\n", file, err)
```

Comment:
This change is not needed now, right?
You return nil in this case.
And there still can be some fatal errors.

Reply:
As noted above, `ctx.generateFile()` returns an error if `ctx.createSyzTest()` or `ctx.createCTest()` do (in fact the latter never returns an error).
Do you think a failure of a single test should be silently aborting the whole process?

Reply:
Is it silent? Don't we print the error on failure?

Reply:
I don't think so, because the error code of generatePrograms is never checked.

In fact I started debugging this because syz-manager printed the following log message:

```
...
exit1 none/thr C                      : OK
exit1 none/thr/repeat C               : OK
ok: 29, broken: 0, skip: 9, fail: 49
2024/07/24 18:00:52 [FATAL] tests failed
```

, from which it wasn't completely clear that there were tests that couldn't be parsed.

Reply:
@dvyukov so what do you think?

Reply:
An error should either fail the whole process, or fail a single test. Failing the whole process for infrastructure failures is fine (say, if we can't read contents of all test files due to permissions, or there is programming error). But I am find if we fail the single test as well.

But both the current code and your new code ignores the error (logging in a test that prints hundreds of lines of output is effectively ignoring).
So I would say either propagate the error and fail everything, or create the failing test here instead of the special case in generateFile.

Reply:
Hm, okay, I like this even better.